cmd/otel-allocator/watcher/file.go

```diff
@@ -43,19 +43,20 @@ func NewFileWatcher(logger logr.Logger, config config.CLIConfig) (*FileWatcher,
 
 	return &FileWatcher{
 		logger:         logger,
-		configFilePath: *config.ConfigFilePath,
+		configFilePath: configFilePath,
 		watcher:        fileWatcher,
 		closer:         make(chan bool),
 	}, nil
 }
 
 func (f *FileWatcher) LoadConfig(_ context.Context) (*promconfig.Config, error) {
```

Comment:
the plan is to deprecate this whole file right? if so, can we add a deprecation notice?

Reply:
I'd rather do it in a separate PR and clearly call it out in the changelog. This one already has a enough changes in it imo.